### PR TITLE
Lpi programming update

### PR DESCRIPTION
**Pull request by djagt**
This branch introduces the `lpiprogram` function, declaring an LPI program structure `prog` as
```
prog = lpiprogram(vars, dom, decvars);
```
with `vars` an Nx2 array specifying the primary spatial variables (`vars(:,1)`) and associated dummy variables (`vars(:,2)`), `dom` an Nx2 array specifying the interval on which each of the spatial variables exists, and `decvars` an optional array of decision variables that appear in the program. For the most part, the function acts as a shell for `sosprogram`, aside from the fact that the domain of the independent variables must be specified as well, and that spatial variables and dummy variables are ordered in a particular order in the program structure (namely `prog.vartable = [vars(:,1);vars(:,2)]`). 
The functions `poslpivar` and `lpivar` (as well as 2D versions) have also been updated to no longer require the spatial domain of the desired operator variable to be declared, but rather, this domain as well as the spatial and dummy variables are now extracted from the LPI program structure fields `prog.vartable` and `prog.dom`. 

In addition, an "LPI" version of a bunch of SOSTOOLS functions has been added, such as `lpidecvar`, `lpisetobj`, and `lpigetsol`, all of which mostly just act as shells for the associated SOSTOOLS functions. However, `lpigetsol` also allows solved values of PI operator decision variables ('dopvar' objects) to be extracted, in that context acting as shell for `getsol_lpivar`. Similarly, `lpi_eq` and `lpi_ineq` (and 2D versions) have been updated to now allow polynomial (rather than operator) equality and inequality constraints to be passed as well, in that case acting as shell for `soseq` and `sosineq`, though practical applications of such functionality may be limited.

Finally, the old version of `lpisolve`, acting as a wrapper function for the pre-defined LPI executives, has now been renamed `lpiscript`. The new function `lpisolve` now acts as a shell for `sossolve`, taking an unsolved LPI program structure and returning the solved program structure.

All the executives and demos have been updated to use the new LPI programming functions. The manual has been updated as well, but the "How_to_get_started" has not been updated yet.

### CODE REVIEW

**Comment by djagt** [comment]
All updates have been tested on all of the PDE examples and some edge cases, but if either @DanBraghini or @sachin-shivakumar  could run some tests as well that would be great. Please let me know if you encounter any issues, and feel free make changes yourself as well of course, or to merge if you approve.

File: PIETOOLS/PIETOOLS_demos/DEMO1_Simple_Stability_Simulation_and_Control.m [hunk 1]
```diff
@@ -1,77 +1,102 @@
...
 ndiff = [0,2,0];    % The PDE state involves 2 first order differentiable state variables   
 
-% To simulate the solution to the PDE system without controller we can use
-% the following code. 
+% Set initial conditions and disturbance for simulation
+syms st sx;
 uinput.ic.PDE = [0,0];  
 uinput.ic.ODE = 0;  
-uinput.u=0;
+uinput.u = 0;
```
Comment (DanBraghini) [inline PIETOOLS/PIETOOLS_demos/DEMO1_Simple_Stability_Simulation_and_Control.m:79]: seems that PIESIM now requires 0*st and 0*sx for uinput to be symbolic. However this throws a warning...

**Review by DanBraghini — APPROVED** [review]

File: PIETOOLS/PIETOOLS_demos/DEMO1_Simple_Stability_Simulation_and_Control.m (hunk 1 above)
Comment (djagt) [reply PIETOOLS/PIETOOLS_demos/DEMO1_Simple_Stability_Simulation_and_Control.m:79]: My issue with the the optimal controller synthesis does seem to be limited to this branch (it does not occur in master), so I'll need to fix that before merging.
